qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -130,79 +140,96 @@ def __init__(self, name, address=None, port=None,
-                            set_cmd=lambda x: self.current_ramp_limit(x),
-                            scale=1/float(self.ask("COIL?")),
-                            unit="T/s")
+                           get_cmd=lambda: self.current_ramp_limit(),
+                           set_cmd=lambda x: self.current_ramp_limit(x),
+                           scale=1/float(self.ask("COIL?")),
+                           unit="T/s")
         if current_ramp_limit is None:
-            self._update_ramp_rate_limit(AMI430._DEFAULT_CURRENT_RAMP_LIMIT, update=False)
+            self._update_ramp_rate_limit(AMI430._DEFAULT_CURRENT_RAMP_LIMIT,
+                                         update=False)
         else:
             self._update_ramp_rate_limit(current_ramp_limit, update=False)
 
         # Add solenoid parameters
         self.add_parameter('coil_constant',
-                            get_cmd=self._update_coil_constant,
-                            set_cmd=self._update_coil_constant,
-                            vals=Numbers(0.001, 999.99999))
-        self.add_parameter('current_rating',
-                            get_cmd="CURR:RATING?",
-                            get_parser=float,
-                            set_cmd="CONF:CURR:RATING {}",
-                            unit="A",
-                            vals=Numbers(0.001, 9999.9999))
-        self.add_parameter('field_rating',
-                            get_cmd=lambda: self.current_rating(),
-                            set_cmd=lambda x: self.current_rating(x),
-                            scale=1/float(self.ask("COIL?")))
+                           get_cmd=self._update_coil_constant,
+                           set_cmd=self._update_coil_constant,
+                           vals=Numbers(0.001, 999.99999))
+
+        # TODO: Not all AMI430s expose this setting. Currently, we
+        # don't know why, but this most likely a firmware version issue,
+        # so eventually the following condition will be smth like
+        # if firmware_version > XX
+        if has_current_rating:
+            self.add_parameter('current_rating',
+                               get_cmd="CURR:RATING?",
+                               get_parser=float,
+                               set_cmd="CONF:CURR:RATING {}",
+                               unit="A",
+                               vals=Numbers(0.001, 9999.9999))
+
+            self.add_parameter('field_rating',
+                               get_cmd=lambda: self.current_rating(),
+                               set_cmd=lambda x: self.current_rating(x),
+                               scale=1/float(self.ask("COIL?")))
+
+        self.add_parameter('current_limit',
+                           unit="A",
+                           set_cmd="CONF:CURR:LIMIT {}",
+                           get_cmd='CURR:LIMIT?',
+                           get_parser=float,
+                           vals=Numbers(0, 80))  # what are good numbers here?
```

Comment:
Should we add field_limit same as field_rating ?

Reply:
Ah, yes, of course. My bad.